app/services/status_message_creation_service.rb

```diff
@@ -54,13 +58,20 @@ def add_photos(status_message, photos)
     end
   end
 
-  def process(status_message, aspect_ids, services)
-    add_to_streams(status_message, aspect_ids) unless status_message.public
+  def validate_aspects
+    raise BadAspectsIDs if aspects.empty?
```

Comment:
I would move this inside the `load_aspects_from_param` (`load_aspects`) method, so you automatically validate it after load. I also think it simplifies it with fewer methods. And that way you can also simplify the block at line 12 to a oneliner.

Reply:
Well, to my taste having more little methods that are doing single thing is a simpler design than having fewer methods doing more :)

I would move the call of  `validate_aspects` to `load_aspects` but leave the method itself unless you're strong about the deletion.

Reply:
> Well, to my taste having more little methods that are doing single thing is a simpler design

It depends. When the method is as simple and short as this one and is only used once, this line could simple replace the line where this method is called. Because then you would save 4 lines which makes the file shorter which increases overall clarity in my opinion.

When the method is used multiple times and/or is a complex line where the method name can simply describe what it does, then I agree with you.

But I think that's not the case here. This method will never be called twice, because it makes only sense to call it at his one point (after loading the aspects) and when you move this to the `load_aspects` method, that is clear and you could even use `@aspects` directly so it's obvious that this validates the just loaded aspects and doesn't call a magic getter that hopefully was loaded before. So it should **never** be called before the `load_aspects` and it doesn't make sense to call it a second time after it, and I think the simplest way to signal this is to just inline the code to `load_aspects` (and the `load_apsect` is then still simple enough even with this code included).

However, if you still think it's better to have this in a separate method, leave it that way, I won't block on that.

Reply:
Okay, it makes sense to me :+1: 

I'll inline the code.